exchange/src/wrapper/algo_wrapper/binary/binary_wrapper.hpp

```diff
@@ -0,0 +1,53 @@
+
+#pragma once
+
+#include "wrapper/algo_wrapper/wrapper.hpp"
+
+namespace nutc {
+namespace wrapper {
+
+class BinaryWrapper : public Wrapper {
+public:
+    BinaryWrapper(
+        std::string algo, std::string trader_id, MarketOrderFunc publish_market_order
+    );
+
+    ~BinaryWrapper() override;
+
+    void fire_on_trade_update(
+        std::string ticker, std::string side, double price, double quantity
+    ) const override;
+
+    void fire_on_orderbook_update(
+        std::string ticker, std::string side, double price, double quantity
+    ) const override;
+
+    void fire_on_account_update(
+        std::string ticker, std::string side, double price, double quantity,
+        double buyer_capital
+    ) const override;
+
+private:
+    class StrategyObject;
```

Comment:
This may be a rare case where forward declaration is required, but we should still discuss the implications of this